src/main/java/dev/openfeature/sdk/OpenFeatureClient.java

```diff
@@ -157,6 +144,31 @@ private <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key
         return details;
     }
 
+    /**
+     * Merge hook and invocation contexts with API, transaction and client contexts.
+     *
+     * @param hookContext       hook context
+     * @param invocationContext invocation context
+     * @return merged evaluation context
+     */
+    private EvaluationContext mergeEvaluationContext(
+            EvaluationContext hookContext,
+            EvaluationContext invocationContext) {
+        final EvaluationContext apiContext = openfeatureApi.getEvaluationContext() != null
+                ? openfeatureApi.getEvaluationContext()
+                : new ImmutableContext();
+        final EvaluationContext clientContext = this.getEvaluationContext() != null
+                ? this.getEvaluationContext()
+                : new ImmutableContext();
+        final EvaluationContext transactionContext = openfeatureApi.getTransactionContext() != null
+                ? openfeatureApi.getTransactionContext()
+                : new ImmutableContext();
+
+        EvaluationContext mergedInvocationCtx = invocationContext.merge(hookContext);
+
+        return apiContext.merge(transactionContext.merge(clientContext.merge(mergedInvocationCtx)));
```

Comment:
Minor - just to match with the spec definition of merging order

```suggestion
  return apiContext.merge(transactionContext.merge(clientContext.merge(invocationContext.merge(hookContext))));
```